corehq/apps/es/client.py

```diff
@@ -308,20 +324,28 @@ class ElasticDocumentAdapter(BaseAdapter):
 
     Subclasses must define the following:
 
-    - ``_index_name``: class attribute (``str``)
-    - ``type``: class attribute (``str``)
-    - ``mapping``: class attribute (``dict``)
+    - ``mapping``: attribute (``dict``)
     - ``from_python(...)``: classmethod for converting models into Elastic format
     """
 
-    @classproperty
-    def index_name(cls):
-        try:
-            return cls.__index_name
-        except AttributeError:
-            prefix = TEST_DATABASE_PREFIX if settings.UNIT_TESTING else ""
-            cls.__index_name = f"{prefix}{cls._index_name}"
-        return cls.__index_name
+    def __init__(self, index_name, type_):
+        """A document adapter for a single index.
+
+        :param index_name: the name of the index that this adapter interacts with
+        :param type_: the index ``_type`` for the mapping
+        """
+        super().__init__()
+        self.index_name = index_name
+        self.type = type_
+
+    def export_adapter(self):
```

Comment:
I see that this is not used anywhere in this PR. Is it true that we never create adapters with export settings anywhere in the code base?

Reply:
We have a utility that routes the `ESQuery` calls to the new adapter in `corehq.apps.es.transient_util.py`.

For the query requiring export adapter, the `doc_adapter_from_info`  in [elastic.py#L60](https://github.com/dimagi/commcare-hq/blob/jm+ap/elastic-multiplexer/corehq/elastic.py#L60 )
 returns the correct adapter.

If you look at [transient_util.py#L50](https://github.com/dimagi/commcare-hq/blob/jm%2Bap%2Felastic-multiplexer/corehq/apps/es/transient_util.py#L50) it makes use of `export_adapter` that you have mentioned